src/parse/text/simple_extensions/argument.rs

```diff
@@ -0,0 +1,633 @@
+                "description".to_string(),
+            )),
+            _ => Ok(description),
+        }
+    }
+}
+
+impl<C: Context> Parse<C> for simple_extensions::ArgumentsItem {
+    type Parsed = ArgumentsItem;
+    type Error = ArgumentsItemError;
+
+    fn parse(self, ctx: &mut C) -> Result<Self::Parsed, Self::Error> {
+        match self {
+            simple_extensions::ArgumentsItem::EnumerationArg(arg) => Ok(ctx.parse(arg)?.into()),
+            simple_extensions::ArgumentsItem::ValueArg(arg) => Ok(ctx.parse(arg)?.into()),
+            simple_extensions::ArgumentsItem::TypeArg(arg) => Ok(ctx.parse(arg)?.into()),
+        }
+    }
+}
+
+impl From<ArgumentsItem> for simple_extensions::ArgumentsItem {
+    fn from(value: ArgumentsItem) -> Self {
+        match value {
+            ArgumentsItem::EnumArgument(arg) => arg.into(),
+            ArgumentsItem::ValueArgument(arg) => arg.into(),
+            ArgumentsItem::TypeArgument(arg) => arg.into(),
+        }
+    }
+}
+
+/// Parse errors for [simple_extensions::ArgumentsItem].
+#[derive(Debug, Error, PartialEq)]
+pub enum ArgumentsItemError {
+    /// Invalid enumeration options.
+    #[error("invalid enumeration options: {0}")]
+    InvalidEnumOptions(#[from] EnumOptionsError),
+
+    /// Empty optional field.
+    #[error("the optional field `{0}` is empty and should be removed")]
+    EmptyOptionalField(String),
+}
+
+/// Arguments that support a fixed set of declared values as constant arguments.
+#[derive(Clone, Debug, PartialEq)]
+pub struct EnumerationArg {
+    /// A human-readable name for this argument to help clarify use.
+    name: Option<String>,
+
+    /// Additional description for this argument.
+    description: Option<String>,
+
+    /// List of valid string options for this argument.
+    options: EnumOptions,
+}
+
+impl<C: Context> Parse<C> for simple_extensions::EnumerationArg {
+    type Parsed = EnumerationArg;
+
+    type Error = ArgumentsItemError;
+
+    fn parse(self, ctx: &mut C) -> Result<EnumerationArg, ArgumentsItemError> {
+        Ok(EnumerationArg {
+            name: ArgumentsItem::parse_name(self.name)?,
+            description: ArgumentsItem::parse_description(self.description)?,
+            options: ctx.parse(self.options)?,
+        })
+    }
+}
+
+impl From<EnumerationArg> for simple_extensions::EnumerationArg {
+    fn from(value: EnumerationArg) -> Self {
+        simple_extensions::EnumerationArg {
+            name: value.name,
+            description: value.description,
+            options: value.options.into(),
+        }
+    }
+}
+
+impl From<EnumerationArg> for simple_extensions::ArgumentsItem {
+    fn from(value: EnumerationArg) -> Self {
+        simple_extensions::ArgumentsItem::EnumerationArg(value.into())
+    }
+}
+
+impl From<EnumerationArg> for ArgumentsItem {
+    fn from(value: EnumerationArg) -> Self {
+        ArgumentsItem::EnumArgument(value)
+    }
+}
```

Comment:
Out of curiosity: are these impls used/required?

Reply:
These are not required but we can make the type conversion simpler with them.

The former is used in:
```rust
impl From<ArgumentsItem> for simple_extensions::ArgumentsItem {
    fn from(value: ArgumentsItem) -> Self {
        match value {
            ArgumentsItem::EnumArgument(arg) => arg.into(),
            ArgumentsItem::ValueArgument(arg) => arg.into(),
            ArgumentsItem::TypeArgument(arg) => arg.into(),
        }
    }
}
```
While the latter is used in:
```rust
impl<C: Context> Parse<C> for simple_extensions::ArgumentsItem {
    type Parsed = ArgumentsItem;
    type Error = ArgumentsItemError;

    fn parse(self, ctx: &mut C) -> Result<Self::Parsed, Self::Error> {
        match self {
            simple_extensions::ArgumentsItem::EnumerationArg(arg) => Ok(ctx.parse(arg)?.into()),
            simple_extensions::ArgumentsItem::ValueArg(arg) => Ok(ctx.parse(arg)?.into()),
            simple_extensions::ArgumentsItem::TypeArg(arg) => Ok(ctx.parse(arg)?.into()),
        }
    }
}
```